docs/RFCS/freeze.md

```diff
@@ -0,0 +1,90 @@
+- Feature Name: Data/network freeze
+- Status: draft
+- Start Date: 2016-02-18
+- Authors: Ben Darnell
+- RFC PR:
+- Cockroach Issue:
+
+
+# Summary
+
+This RFC outlines the plan for freezing our data formats and network
+protocols.
+
+# Motivation
+
+We currently make backwards-incompatible changes to data formats
+without providing any means to upgrade without data loss. This will
+need to stop before beta for obvious reasons.
```

Comment:
I considered adding a "stage 1.5" in which migration would be possible but would have to go through some sort of dump/restore tool. I dropped it because I thought that anything the dump/restore tool could do could also be done at store startup (although on further reflection there are some things that would probably be easier with this process).

Is it worth providing this guarantee in beta? That's one of the reasons why this is an RFC. My opinion is that the guarantees discussed here are an essential part of the difference between alpha and beta. We need our beta users to start putting cockroachdb into production-like scenarios, and delaying the point at which we reach data-format stability delays the point at which we can get useful feedback.